mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/Client.h

```diff
@@ -85,6 +89,19 @@ class CompilationTask : public CompilationTaskBase {
 // CompilerClient
 //===----------------------------------------------------------------------===//
 
+class CompilerClient;
+
+using OptionsConstructorFuncT =
+    std::function<StatusOr<std::unique_ptr<mlir::OptionsContext>>(
+        const CompilerClient &client, const llvm::ArrayRef<llvm::StringRef>)>;
+
+void registerOption(const llvm::StringRef optionsType,
+                    OptionsConstructorFuncT func);
+
+StatusOr<std::unique_ptr<mlir::OptionsContext>>
+createOptions(const CompilerClient &client, const llvm::StringRef optionsType,
+              const llvm::ArrayRef<llvm::StringRef> args);
+
```

Comment:
Do we want to move all of this to a separate header? There's really nothing tying it to the client anymore.